src/data/nat/basic.lean

```diff
@@ -626,6 +626,13 @@ begin
   simp [add_comm, nat.add_sub_assoc, one_le_iff_ne_zero.2 hi]
 end
 
+lemma le_succ_mul_neg (n : ℕ) : ∀ d, d ≤ (n + 1) * d - n
```

Comment:
```suggestion
lemma le_succ_mul_sub (n : ℕ) : ∀ d, d ≤ (n + 1) * d - n
```

Reply:
I don't have strong feelings here but I have some reservations about this lemma because:

1. it's _very_ specific
2. it uses nat subtraction

Are you sure we want it?

Reply:
I agree it's very specific - and the construction here only needs `d = 2` but it does need nat subtraction for this application

Reply:
Actually we only need the case `n = 2` instead. The statement for `d = 0` is the entire point of this lemma and this what makes it difficult to inline, so I first made it a private lemma, and subsequently decided that we didn't like private lemmas so I would better be off giving it a proper name and putting it in the right file instead.

Reply:
(oops sorry I meant `n = 2`, and indeed the `d = 0` is precisely the important case for this)

Reply:
I'm confused by the claims about `d = 0` being important above. For the record, there are two applications of this lemma and they both solve :
```
n d : ℕ
hd: 0 < d
⊢ d ≤ 2 * d - 1
```
I still think we shouldn't need weirdly-specific lemmas like this but I'm happy to let this point go.

I'll add a remark about how we might avoid needing this lemma elsewhere.